app/src/androidTest/java/sw19/moring03/paint/UndoButtonTest.java

```diff
@@ -32,48 +34,43 @@
 
     @Test
     public void testButtonVisibility() {
-        onView(withId(R.id.undoButton)).check(doesNotExist());
         onView(withId(R.id.toolChooserButton)).perform(click());
         onView(withId(R.id.drawPointButton)).perform(click());
         onView(withId(R.id.drawingView)).perform(touchAt(400, 400));
-
+        onView(withId(R.id.undoButton)).check(matches((isEnabled())));
         onView(withId(R.id.undoButton)).perform(click());
     }
 
     @Test
     public void testUndoLastObject() {
-        onView(withId(R.id.undoButton)).check(doesNotExist());
 
         onView(withId(R.id.toolChooserButton)).perform(click());
         onView(withId(R.id.drawPointButton)).perform(click());
         onView(withId(R.id.drawingView)).perform(touchAt(400, 400));
+        onView(withId(R.id.undoButton)).check(matches((isEnabled())));
```

Comment:
is this necessary, since the next line tries to click on the undo button and should if not enabled throw an error